src/libsync/creds/oauth.cpp

```diff
@@ -718,4 +723,20 @@ void AccountBasedOAuth::refreshAuthentication(const QString &refreshToken)
     });
 }
 
+
+bool OAuth::checkForRedirctError(QNetworkReply *reply)
+{
+    // Special handling for BIGIP F5, the apm might redirect us at random to present us we new cookies.
+    // this redirect during a refresh is supposed to trigger a retry, where the connection validator is run again.
+    if (reply->attribute(QNetworkRequest::RedirectionTargetAttribute).isNull()) {
+        qCWarning(lcOauth) << "redirects during authentication are not supported";
+        if (_isRefreshingToken) {
+            Q_EMIT refreshError(QNetworkReply::TooManyRedirectsError, tr("Unsupported redirect on %1").arg(wellKnownPathC));
```

Comment:
Not a fan of just using this error, because we can't tell for sure in the caller whether this was due to this issue. Sure, at this point, we don't have to, and the error is logged... Well, good enough for now, I guess...